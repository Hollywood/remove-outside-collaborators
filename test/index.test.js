const {createRobot} = require('probot')
const plugin = require('../index')

describe('plugin', () => {
  let robot
  let event
  let analyze

  beforeEach(() => {
    robot = createRobot()
    robot.auth = () => Promise.resolve({})

    analyze = jest.fn()

    plugin(robot, {}, { analyze })
  })

  describe('analyzes on added collaborator', () => {
    beforeEach(() => {
      event = {
        event: 'member',
        payload: JSON.parse(JSON.stringify(require('./fixtures/events/collaborator.added.json')))
      }
    })
    it('analyzes member', async () => {
      await robot.receive(event)
      expect(analyze).toHaveBeenCalled()
    })
  })
})
