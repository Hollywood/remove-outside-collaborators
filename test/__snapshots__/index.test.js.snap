// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin opens an issue 1`] = `
Object {
  "body": "<h3 align=\\"center\\">NOTE: Outside Collaborators cannot be added to this Org. Please contact an Admin to override.</h3>
<p align=\\"center\\">Collaborator added: <strong>hollywood</strong></p>

---

",
  "owner": "hollywood",
  "repo": "test",
  "title": "[CRITICAL] An Outside Collaborator has been removed!",
}
`;
